src/async_support.rs

```diff
@@ -53,6 +53,19 @@ impl Read for AsyncReadWrapper {
     }
 }
 
+impl Seek for AsyncReadWrapper {
+    fn seek(&mut self, seek: SeekFrom) -> std::io::Result<u64> {
+        if self.rx.is_terminated() {
+            return Ok(0);
+        }
+        Ok(match seek {
+            SeekFrom::Start(n) => n,
+            SeekFrom::Current(n) => n as u64,
+            SeekFrom::End(n) => n as u64,
+        })
+    }
+}
```

Comment:
This can be replaced with:
```rust
impl Seek for AsyncReadWrapper {
    fn seek(&mut self, seek: SeekFrom) -> std::io::Result<u64> {
        Ok(0)
    }
}
```

This hints Rust compiler that the seek is indeed supported, but underlying, it is done by the `libarchive_seek_callback()` callback. This also deserves a comment block explaining this.

Reply:
Sure, i'll done that.

Reply:
This code is not called so we could do even better, as:
```rust
// Hints Rust compiler that the seek is indeed supported, but
// underlying, it is done by the libarchive_seek_callback() callback.
impl Seek for AsyncReadWrapper {
    fn seek(&mut self, seek: SeekFrom) -> std::io::Result<u64> {
        unreachable!("We need to use libarchive_seek_callback() underlying.")
    }
}
```